spring-cloud-gcp-samples/spring-cloud-gcp-data-spanner-sample/src/main/java/com/example/SpannerExampleDriver.java

```diff
@@ -0,0 +1,52 @@
+package com.example;
+
+import java.util.Arrays;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.ApplicationRunner;
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Profile;
+
+@SpringBootApplication
+public class SpannerExampleDriver {
+
+	private static final Log LOGGER = LogFactory.getLog(SpannerExampleDriver.class);
+
+	@Autowired
+	private SpannerRepositoryExample spannerRepositoryExample;
+
+	@Autowired
+	private SpannerTemplateExample spannerTemplateExample;
+
+	public static void main(String[] args) {
+		System.out.println(Arrays.toString(args));
+		SpringApplication.run(SpannerExampleDriver.class, args);
+	}
+
+	@Bean
+	@Profile("!test")
```

Comment:
What's this profile for?

Reply:
The `@SpringBootTest` annotation picks up any ApplicationRunner or CommandLineRunner bean and automatically runs it. However, in test we want to control when it gets run; we want to run the sample after we made some assertions. This allows us to first assert the Spanner tables are empty, run the sample, then verify the rows were created. 

`@Profiles` lets us exclude the bean from the Test if the `Test` profile is set, so that the bean does not automatically get run.

Reference: https://stackoverflow.com/questions/29344313/prevent-application-commandlinerunner-classes-from-executing-during-junit-test